src/jdk.incubator.jextract/share/classes/jdk/internal/clang/Diagnostic.java

```diff
@@ -72,12 +73,16 @@ public int severity() {
     }
 
     public SourceLocation location() {
-        return new SourceLocation(Index_h.clang_getDiagnosticLocation(ptr));
+        try (ResourceScope scope = ResourceScope.newConfinedScope()) {
+            return new SourceLocation(Index_h.clang_getDiagnosticLocation(scope, ptr));
```

Comment:
SourceLocation holds a reference to the segment, but this closes the scope after this call. Looks like this should use an implicit scope instead.